apps/web/src/components/Solana/useWalletModal.tsx

```diff
@@ -0,0 +1,36 @@
+"use client";
+
+import { createContext, useContext } from "react";
+
+export interface WalletModalContextState {
+  visible: boolean;
+  setVisible: (open: boolean) => void;
+}
+
+const DEFAULT_CONTEXT = {
+  setVisible(_open: boolean) {
+    console.error(constructMissingProviderErrorMessage("call", "setVisible"));
+  },
+  visible: false,
+};
+Object.defineProperty(DEFAULT_CONTEXT, "visible", {
+  get() {
+    console.error(constructMissingProviderErrorMessage("read", "visible"));
+    return false;
+  },
+});
+
+function constructMissingProviderErrorMessage(
```

Comment:
We should use Next.js error handling where possible rather than rolling our own handling. Great to see this in here though!

https://nextjs.org/docs/app/getting-started/error-handling